File: csharp/luhn/Luhn.cs
public static class Luhn
{
    public static bool IsValid(string number)
    {
        string cleanedNumber = number.Replace(" ", "");
        if (cleanedNumber.Length < 2)
        {
            return false;
        }
        int sum = 0;
        bool doubleDigit = false;

        for (int i = cleanedNumber.Length - 1; i >= 0; i--)
        {
            char c = cleanedNumber[i];
            if (!char.IsDigit(c))
            {
                return false;
            }
            int digit = c - '0';
            if (doubleDigit)
            {
                digit *= 2;
                if (digit > 9)
                {
                    digit -= 9;
                }
            }
            sum += digit;
            doubleDigit = !doubleDigit;
        }
        return (sum % 10 == 0);
    }
}

File: javascript/luhn/luhn.js
//
// This is only a SKELETON file for the 'Luhn' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const valid = (string) => {
  const cleanString = string.replace(/\s+/g, '');
  if (cleanString.length <= 1 || /\D/.test(cleanString)) { return false; }
  return [...cleanString].reverse()
                         .map(Number)
                         .reduce((sum, digit, index) => {
                          if (index % 2 === 1) {
                            const doubled = digit * 2;
                            return sum + (doubled > 9 ? doubled - 9 : doubled);
                          }
                          return sum + digit;
                         }, 0) % 10 === 0;
};
